core/src/main/scala/org/apache/spark/rapids/tool/benchmarks/Benchmark.scala

```diff
@@ -101,20 +103,30 @@ class Benchmark(
     // The results are going to be processor specific so it is useful to include that.
     out.println(RuntimeUtil.getJVMOSInfo.mkString("\n"))
     val nameLen = Math.max(40, Math.max(name.length, benchmarks.map(_.name.length).max))
-    out.printf(s"%-${nameLen}s %14s %14s %11s %10s\n",
-      name + ":", "Best Time(ms)", "Avg Time(ms)", "Stdev(ms)", "Relative")
+    out.printf(s"%-${nameLen}s %14s %14s %11s %10s %10s %10s\n",
+      name + ":", "Best Time(ms)", "Avg Time(ms)", "Stdev(ms)","Max GC Time(ms)",
+      "Max GC Count", "Relative")
     out.println("-" * (nameLen + 80))
     results.zip(benchmarks).foreach { case (result, benchmark) =>
-      out.printf(s"%-${nameLen}s %14s %14s %11s %10s\n",
+      out.printf(s"%-${nameLen}s %14s %14s %11s %10s %10s %10s\n",
         benchmark.name,
         "%5.0f" format result.bestMs,
         "%4.0f" format result.avgMs,
         "%5.0f" format result.stdevMs,
+        "%8d" format result.maxGcTime,
+        "%5d" format result.maxGcCount,
         "%3.1fX" format (firstBest / result.bestMs))
     }
     out.println()
   }
 
+  private def getGcMetrics: (Any, Any) = {
+    val gcBeans = ManagementFactory.getGarbageCollectorMXBeans
+    val gcTime = gcBeans.map(_.getCollectionTime).sum
+    val gcCount = gcBeans.map(_.getCollectionCount).sum
+    (gcTime, gcCount)
+  }
+
```

Comment:
If we move this to the `RuntimeUtil`, then we can use any where as a utility function regardless it is part of the benchmark or not.
Also, we can return a case class

```
case class JVMGCInfo(val gcCount: Long, val gcTime: Long, val: totalHeap: Long, val usedHeap: Long, val freeHeap: Long)
```

Reply:
Added changes to handle the above. Added memory tracker class. Also added tracking of average used memory, average free memory and max heap allocation